### PR TITLE
Sleeping changes

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

End-user summary- If you sleep on a bed with sheets and are not in crit, you'll heal 0.25 brute, 0.25 burn, 0.125 toxloss, 25 stamina, and a variable amount of cloneloss (-0.1 to 0.2, depending on your overall health). If you sleep on the floor, you get a quarter of those values.

First off, this uses regular ticks, not lifeticks, so there's that. You heal 20 times in the span that chems would heal you once, so let me just get that outta the way. The numbers are bigger than they look to be.

Adjusts health_ratio number to be -0.5 or greater, hence you'll heal if sleeping above crit. If you're crit, you're SOL, and probably also getting eaten by a ravager or something.

Does fancy math to standardize the base healing amount to 0.25 per two seconds when in a bed with a bedsheet on it. Let's see... we want to divide that by 20 to get regular gameticks, so 0.0125, then divide *that* by 4 for 0.003125, and assign as needed to bed, table, floor, and bedsheet. Easy.

So now a standard bed with a bedsheet will heal a base amount of 0.25/2 second interval. Brute and burn stay as is, toxloss stays as is, staminaloss is multiplied by 100 (25, regular staminaheal is usually *higher* than that!), and cloneloss... cloneloss is adjusted to a base healrate of 0.2/ 2 seconds, at 100% hp.

Additionally, the cloneloss recovery rate is tied directly to how healthy you are via your health ratio. This means that if you go to sleep below 0 HP, or keep getting slashed in and out of crit and get unlucky on tickrate, you can end up with trace amounts of cloneloss. Just take a brief nap once healed if it really bothers you to have 0.5835 cloneloss, it'll heal within seconds.

Drunkenness left untouched.

## Why It's Good For The Game

Fixes a broken system, RP uses, way to slowly heal a small amount of cloneloss planetside. Really, mostly RP uses.

## Changelog
:cl:
expansion: Added stuff to sleeping. If you sleep on a bed with sheets and are not in crit, you'll heal 0.25 brute, 0.25 burn, 0.125 toxloss, 25 stamina, and a variable amount of cloneloss (-0.1 to 0.2, depending on your overall health). If you sleep on the floor, you get a quarter of those values.
/:cl:

### CODE REVIEW

**Review by BraveMole — CHANGES_REQUESTED** [review]
More descriptive cl please, like sleeping heal when you're in worst condition or whatever

**Comment by EvilMelonTheSecond** [comment]
Thank you for buffing my frontline bed build. Now I can convince people to take laundry areas for me to get bedsheets

**Comment by jroinc** [comment]
> 
> 
> More descriptive cl please, like sleeping heal when you're in worst condition or whatever

It's pretty dammed descriptive, but I can add some more end-user stuff, I guess.

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -126,21 +126,22 @@
...
-			healing -= 0.1
+			healing -= 0.003125
 		for(var/obj/item/bedsheet/bedsheet in range(owner.loc,0))
 			if(bedsheet.loc != owner.loc) //bedsheets in your backpack/neck don't give you comfort
 				continue
-			healing -= 0.1
+			healing -= 0.003125
 			break //Only count the first bedsheet
-		if(health_ratio > 0.8)
+		if(health_ratio > -0.5)
```
Comment (TiviPlus) [inline code/datums/status_effects/debuffs.dm:139]: All of these changed number are magic numbers and need to be behind defines

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Your CL is insufficiently descriptive of the changes implemented please add a more player-friendly changelog
(in the realm of "x is now heals more, y will now make you less drowsy", etc)

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (jroinc) [reply code/datums/status_effects/debuffs.dm:139]: As... opposed to the previous magic numbers?

Comment (jroinc) [reply code/datums/status_effects/debuffs.dm:139]: Why? What's the gain? I can do the math in the var easy enough, but it's used just one place AFAIK.

**Comment by jroinc** [comment]
> 
> 
> Your CL is insufficiently descriptive of the changes implemented please add a more player-friendly changelog
> (in the realm of "x is now heals more, y will now make you less drowsy", etc)

You mean it's *too* descriptive.

End-user summary- If you sleep on a bed with sheets and are not in crit, you'll heal 0.25 brute, 0.25 burn, 0.125 toxloss, 25 stamina, and a variable amount of cloneloss (-0.1 to 0.2, depending on your overall health). If you sleep on the floor, you get a quarter of those values.

**Comment by BraveMole** [comment]
image

acording to you, this is too descriptive?

**Comment by jroinc** [comment]
> 
> 
> image
> 
> acording to you, this is too descriptive?

ohhhh you mean the... Okay, I see. One sec.

**Review by BraveMole — APPROVED** [review]

**Pull request merged**